sdk/tests/deqp/framework/opengl/gluTextureUtil.js

```diff
@@ -0,0 +1,1018 @@
+    }
+
+    switch (texFormat.type)
+    {
+        case textureFormat.ChannelType.SNORM_INT8: type = gl.BYTE; break;
+        case textureFormat.ChannelType.SNORM_INT16: type = gl.SHORT; break;
+        case textureFormat.ChannelType.UNORM_INT8: type = gl.UNSIGNED_BYTE; break;
+        case textureFormat.ChannelType.UNORM_INT16: type = gl.UNSIGNED_SHORT; break;
+        case textureFormat.ChannelType.UNORM_SHORT_565: type = gl.UNSIGNED_SHORT_5_6_5; break;
+        case textureFormat.ChannelType.UNORM_SHORT_4444: type = gl.UNSIGNED_SHORT_4_4_4_4; break;
+        case textureFormat.ChannelType.UNORM_SHORT_5551: type = gl.UNSIGNED_SHORT_5_5_5_1; break;
+        case textureFormat.ChannelType.SIGNED_INT8: type = gl.BYTE; break;
+        case textureFormat.ChannelType.SIGNED_INT16: type = gl.SHORT; break;
+        case textureFormat.ChannelType.SIGNED_INT32: type = gl.INT; break;
+        case textureFormat.ChannelType.UNSIGNED_INT8: type = gl.UNSIGNED_BYTE; break;
+        case textureFormat.ChannelType.UNSIGNED_INT16: type = gl.UNSIGNED_SHORT; break;
+        case textureFormat.ChannelType.UNSIGNED_INT32: type = gl.UNSIGNED_INT; break;
+        case textureFormat.ChannelType.FLOAT: type = gl.FLOAT; break;
+        case textureFormat.ChannelType.UNORM_INT_101010: type = gl.UNSIGNED_INT_2_10_10_10_REV; break;
+        case textureFormat.ChannelType.UNORM_INT_1010102_REV: type = gl.UNSIGNED_INT_2_10_10_10_REV; break;
+        case textureFormat.ChannelType.UNSIGNED_INT_1010102_REV: type = gl.UNSIGNED_INT_2_10_10_10_REV; break;
+        case textureFormat.ChannelType.UNSIGNED_INT_11F_11F_10F_REV: type = gl.UNSIGNED_INT_10F_11F_11F_REV; break;
+        case textureFormat.ChannelType.UNSIGNED_INT_999_E5_REV: type = gl.UNSIGNED_INT_5_9_9_9_REV; break;
+        case textureFormat.ChannelType.HALF_FLOAT: type = gl.HALF_FLOAT; break;
+        case textureFormat.ChannelType.FLOAT_UNSIGNED_INT_24_8_REV: type = gl.FLOAT_32_UNSIGNED_INT_24_8_REV; break;
+        case textureFormat.ChannelType.UNSIGNED_INT_24_8: type = texFormat.order == textureFormat.ChannelType.D ?
+                                                                 gl.UNSIGNED_INT : gl.UNSIGNED_INT_24_8; break;
+
+        default:
+            throw new Error("Can't map texture format to GL transfer format " + textureFormat.type);
+    }
+
+    return new TransferFormat(format, type);
+};
+
+/*--------------------------------------------------------------------*//*!
+ * \brief Map tcu::TextureFormat to GL internal sized format.
+ *
+ * Maps generic texture format description to GL internal format.
+ * If no mapping is found, throws tcu::InternalError.
+ *
+ * \param texFormat Generic texture format.
+ * \return GL sized internal format.
+ *//*--------------------------------------------------------------------*/
+var getInternalFormat = function(/*tcu::TextureFormat*/ texFormat)
+{
+
+    var stringify = function(order, type) {
```

Comment:
This stringify function isn't needed; you can use the PACK_FMT macro as a function similarly to how the C++ code works.

var PACK_FMT = function(ORDER, TYPE) { return ORDER << 16 | TYPE; }

function getInternalFormat(order, type) {
    switch (PACK_FMT(order, type)) {
	case PACK_FMT(TextureFormats.ChannelOrder.RGBA, TextureFormats.ChannelType.UNORM_SHORT_5551): console.log("RGB5_A1"); break;
	case PACK_FMT(TextureFormats.ChannelOrder.RGBA, TextureFormats.ChannelType.UNORM_SHORT_4444): console.log("RGBA4"); break;
	case PACK_FMT(TextureFormats.ChannelOrder.RGB, TextureFormats.ChannelType.UNORM_SHORT_565): console.log("RGB565"); break;
	default: console.log("default case for " + order + " " + type);
    }
}